src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/AbstractMappedMinecraftProvider.java

```diff
@@ -99,9 +106,29 @@ public List<MinecraftJar> provide(ProvideContext context) throws Exception {
 			}
 		}
 
-		return remappedJars.stream()
-				.map(RemappedJars::outputJar)
-				.toList();
+		return minecraftJars;
+	}
+
+	// Create two copies of the remapped jar, the backup jar is used as the input of genSources
+	public static Path getBackupJarPath(MinecraftJar minecraftJar) {
+		final Path outputJarPath = minecraftJar.getPath();
+		return outputJarPath.resolveSibling(outputJarPath.getFileName() + ".backup");
+	}
+
+	protected boolean hasBackupJars(List<MinecraftJar> minecraftJars) {
+		for (MinecraftJar minecraftJar : minecraftJars) {
+			if (!Files.exists(getBackupJarPath(minecraftJar))) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	protected void createBackupJars(List<MinecraftJar> minecraftJars) throws IOException {
+		for (MinecraftJar minecraftJar : minecraftJars) {
+			Files.copy(minecraftJar.getPath(), getBackupJarPath(minecraftJar));
```

Comment:
Needs `REPLACE_EXISTING`, otherwise it'll crash if for example some jars exist (so `hasBackupJars` returns false) and others don't.